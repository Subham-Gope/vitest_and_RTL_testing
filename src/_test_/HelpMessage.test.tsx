// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import HelpMessage from "../components/HelpMessage";

// describe("Testing the HelpMessage.tsx file", () => {
//   test("displays help message onHover", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     // Setting up snapshot matching
//     const { container } = render(<HelpMessage />);
//     expect(container).toMatchSnapshot();

//     const btn = screen.getByRole("button", { name: /help/i });
//     expect(btn).toBeInTheDocument();

//     // Initiating user interaction
//     await user.hover(btn);

//     const helpText = screen.getByText(/need help?/i);
//     expect(helpText).toBeInTheDocument();

//     await user.unhover(btn);
//     expect(helpText).not.toBeInTheDocument();
//   });
// });
