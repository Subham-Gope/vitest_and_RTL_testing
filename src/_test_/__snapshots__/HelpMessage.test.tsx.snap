// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the HelpMessage.tsx file > displays help message onHover 1`] = `
<div>
  <div
    class="min-h-screen w-screen overflow-hidden relative"
  >
    <span
      aria-label="help"
      class="p-4 bg-red-100 border-red-300 border-[0.5px] absolute top-[40%] left-[50%] -translate-x-[50%]"
      role="button"
    >
      ❓
    </span>
    <div>
       
    </div>
  </div>
</div>
`;
